### PR TITLE
Implement ICS 721

### PR DESCRIPTION
**Pull request by yito88**
Import proto from https://github.com/bianjieai/nft-transfer for ICS-721

### CODE REVIEW

**Comment by Farhad-Shabani** [comment]
@yito88 would you regenerate protos given that the proto compiler has been revised as of `v0.40.0` release? Thanks!

File: src/prost/ibc.applications.nft_transfer.v1.serde.rs [hunk 1]
```diff
@@ -0,0 +1,2014 @@
...
+        }
+        if true {
+            len += 1;
+        }
+        if true {
+            len += 1;
+        }
+        let mut struct_ser = serializer.serialize_struct("ibc.applications.nft_transfer.v1.NonFungibleTokenPacketData", len)?;
+        if true {
+            struct_ser.serialize_field("classId", &self.class_id)?;
```
Comment (yito88) [inline src/prost/ibc.applications.nft_transfer.v1.serde.rs:780]: @Farhad-Shabani I think the field name should be snake_case. How can I fix it?

Comment (Farhad-Shabani) [reply src/prost/ibc.applications.nft_transfer.v1.serde.rs:780]: Per our chat, we should be fine to modify the relevant hardcoded strings within the `ibc-rs` tests, so can adhere to the Protobuf JSON encoding guidelines.

**Review by Farhad-Shabani — COMMENTED** [review]
We just need to address the issue with the `cargo doc`. It can be patched automatically the same way we do that for the ICS-20: https://github.com/cosmos/ibc-proto-rs/blob/aecfe0de5983f20fcc3af8f2e2b73384319b89a8/tools/proto-compiler/src/cmd/compile.rs#L225-L231


Once that's resolved, everything else appears good, and we can proceed with merging this PR.

**Review by Farhad-Shabani — APPROVED** [review]
Thank you @yito88, for sorting things out! 🎉

**Pull request merged**